### DISCUSSION TITLE
Possible issue with visreg and extra_time

### DISCUSSION THREAD

**Discussion by Nathan-Hebert** [sdmTMB/sdmTMB#330] [Q&A] [answered]
Currently using sdmTMB_0.4.2.9003. Reproducible example:

_
library(sdmTMB)

pcod$fyear <- as.factor(pcod$year)
mesh <- make_mesh(pcod, c("X", "Y"), cutoff = 20)
fit <- sdmTMB(
    density ~ fyear, time = "year", spatiotemporal = "iid",
    data = pcod, mesh = mesh,
    family = tweedie(link = "log"), extra_time = c(2012)
)
visreg::visreg(fit, "fyear")
_

**Error in data.frame(xy[[j]]$x$D, visregRes = xy[[j]]$y$r, visregPos = xy[[j]]$y$pos) : 
  arguments imply differing number of rows: 2143, 2144
In addition: Warning messages:
1: In Response(fit, x, trans, alpha, ...) :
  Residuals do not match data; have you changed the original data set?  If so, visreg is probably not displaying the residuals for the data set that was actually used to fit the model.
2: In y + rr :
  longer object length is not a multiple of shorter object length**

**Comment by Nathan-Hebert** [comment]
I'm thinking this is connected to the "fake" rows added with extra_time.

**Comment by seananderson** [comment, marked as answer]
Sorry for another extra-time bug. This time, however, I think I've already fixed it as part of a recent overhaul. Can you try updating sdmTMB and seeing if this works for you? I'm using version 0.4.3.9008.

``` r
library(sdmTMB)
  pcod$fyear <- as.factor(pcod$year)
  mesh <- make_mesh(pcod, c("X", "Y"), cutoff = 20)
  fit <- sdmTMB(
    density ~ fyear, 
    time = "year", 
    spatiotemporal = "iid",
    data = pcod,
    mesh = mesh,
    family = tweedie(), 
    extra_time = 2012
  )
  visreg::visreg(fit, "fyear")
```

![](https://i.imgur.com/Pdt6d3D.png)

<sup>Created on 2024-03-27 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Reply by Nathan-Hebert** [reply]
Yep, it's fixed. Thanks!
